### DISCUSSION TITLE
Mesh combined by different type of elements

### DISCUSSION THREAD

**Discussion by tlorentz77** [mfem/mfem#3589] [Q&A]
Hello MFEM community,

Does MFEM support to use different type of element in a single model?

For example, a model has 1D truss elements and 2D shell elements in 3D space.

Thanks!

**Comment by v-dobrev** [comment]
It is possible to express such mesh in the MFEM format -- the 2D elements will be the "elements" in the mesh and the 1D elements will be the "boundary" elements. The dimension of the vertices can be 2 or 3 in this case too. See the description here: https://mfem.org/mesh-format-v1.0.

**Reply by tlorentz77** [reply]
Hi @v-dobrev, thanks for your quick reply.

I appologize for the unclarity. What I want is that 1D truss element and 2D shell element shared a common node, as shown in the following, instead of sharing a common edge.




![image](https://user-images.githubusercontent.com/112360263/230249062-4867e889-b26b-46aa-bf77-f967f3958198.png)

**Reply by termi-official** [reply]
It is possible to implement this, but it is a bit tricky. I implemented something similar, where I have the 1D domain and the 2D domain stored as separated meshes. The solutions are coupled with an iterative scheme (i.e. I do not solve these as one big problem, but two smaller problems which are solved alternatingly). ~~However, you could also build a matrix $P$ which couples your problems after assembly by enforcing equality of the nodal values.~~ I think the new `SubMesh` class could help here, but I had no time to test it with my problems.

Well, its mechanics, so it depends on your specific formulation how you have to coupled the problems.
